subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_rest

```diff
@@ -10,17 +10,31 @@ menuconfig NRF_CLOUD_REST
 
 if NRF_CLOUD_REST
 
-config NRF_CLOUD_REST_HOST_NAME
-	string "nRF Cloud REST API hostname"
-	default "api.nrfcloud.com"
-
-config NRF_CLOUD_REST_FRAGMENT_SIZE
-	int "Fragment size for REST API downloads"
-	range 128 1700
-	default 1700
-
-module = NRF_CLOUD_REST
-module-str = nRF Cloud REST
-source "subsys/logging/Kconfig.template.log_config"
+	config NRF_CLOUD_REST_HOST_NAME
+		string "nRF Cloud REST API hostname"
+		default "api.nrfcloud.com"
+
+	config NRF_CLOUD_REST_AUTOGEN_JWT
+		bool "nRF Cloud REST Auto-generate JWTs"
+		default y
+		select MODEM_JWT
```

Comment:
probably better practice to make this `depends on MODEM_JWT` instead of `select`, since you are defaulting to y

Reply:
In that case, autogen would be enabled or disabled depending on whether another library has MODEM_JWT as a dependency, which could get confusing. I would rather make autogen default to disabled than use `depends on`. 

I'd really like to avoid autogen being disabled by default, I think that runs the risk of obscuring it quite a bit. I'd prefer that advanced configuration be a requirement for optimization, rather than a requirement for QOL.

But you might feel differently -- I'll leave the final decision up to you. `select` and enabled by default, or `depends on` and disabled by default?

Reply:
enabling `NRF_CLOUD_REST` should not also enable `MODEM_JWT`.  
it should be up to the user to enable `MODEM_JWT` if their project needs it.

Reply:
Okay! If you don't mind then, I'll also set this feature to default to disabled.